velox/exec/tests/RowContainerTest.cpp

```diff
@@ -278,7 +279,44 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
   }
 
   template <typename T>
-  void testCompareFloats(TypePtr type, bool ascending, bool nullsFirst) {
+  void setupFloatTestExpectedOutput(
+      T& initialOrder,
```

Comment:
Is T always std::vector<std::optional<U>>? If so, maybe it would be clearer as

```
  template <typename T>
  void setupFloatTestExpectedOutput(
      std::vector<std::optional<T>>& initialOrder,
```

It is not clear what this method does. Maybe add a comment to clarify?

Reply:
I added a comment. 

> Is T always std::vector<std::optional>?

No, they are not. The initial order type changes based on the complex type. In the end these are row, map or array vectors and not only `std::vector<std::optional<T>>` (see the various types for `expectedOrder` in the test functions.
But each time the reordering of the elements based on the test inputs is the same, hence I made a templated version. The exception is the array test. But otherwise, the function is shared for the simple, map and row test. Admittedly, the simple and row test have the same type while map is different.